### PR TITLE
Batch Resource

### PR DESCRIPTION
**Pull request by RonAzar**

### CODE REVIEW

File: src/main/kotlin/com/hibob/academy/resource/PetsResource.kt [hunk 1]
```diff
@@ -75,4 +75,27 @@ class PetsResource (private val petService: PetService){
...
+        @PathParam("companyId") companyId: Long,
+        petIds: List<Long>
+    ): Response {
+        val petsAdoptedCount = petService.adoptMultiplePets(ownerId, petIds, companyId)
+        return Response.ok("$petsAdoptedCount pets were successfully adopted.").build()
+    }
+
+    @POST
+    @Path("/owner/{ownerId}/company/{companyId}/pets/create")
+    fun createMultiplePets(
```
Comment (yaelblsch) [inline src/main/kotlin/com/hibob/academy/resource/PetsResource.kt:92]: Why do you need owner id as path param in multiple pets creation?  They don't have to belong to same owner
In addition isn't owner a part of the pet itself which you can get in the petRecords?

File: src/main/kotlin/com/hibob/academy/resource/PetsResource.kt [hunk 2]
```diff
@@ -75,4 +75,26 @@ class PetsResource (private val petService: PetService){
...
     }
+
+    @POST
+    @Path("/owner/{ownerId}/company/{companyId}/pets/adopt")
+    fun adoptPets(
+        @PathParam("ownerId") ownerId: Long,
+        @PathParam("companyId") companyId: Long,
+        petIds: List<Long>
+    ): Response {
+        val petsAdoptedCount = petService.adoptMultiplePets(ownerId, petIds, companyId)
```
[the remark this answers is not in the document]
Comment (RonAzar) [reply src/main/kotlin/com/hibob/academy/resource/PetsResource.kt]: Fixed

File: src/main/kotlin/com/hibob/academy/resource/PetsResource.kt [hunk 3]
```diff
@@ -75,4 +75,26 @@ class PetsResource (private val petService: PetService){
...
+        return Response.ok("$petsAdoptedCount pets were successfully adopted.").build()
+    }
+
+    @POST
+    @Path("/company/{companyId}/pets/create")
```
[the remark this answers is not in the document]
Comment (RonAzar) [reply src/main/kotlin/com/hibob/academy/resource/PetsResource.kt]: Got it!

File: src/main/kotlin/com/hibob/academy/resource/PetsResource.kt [hunk 4]
```diff
@@ -75,4 +75,25 @@ class PetsResource (private val petService: PetService){
         val petTypeCountMap = petService.getPetTypesAmount(companyId)
         return Response.ok(petTypeCountMap).build()
     }
+
+    @POST
```
Comment (yaelblsch) [inline src/main/kotlin/com/hibob/academy/resource/PetsResource.kt:79]: Adopt is update and not create therefore should be PUT and not POST

**Review by yaelblsch — APPROVED** [review]

**Pull request merged**